### PR TITLE
Prep: move composer type, proving key and verification key.

### PR DESCRIPTION
**Pull request by codygunton**
# Description

The goal here was to make the shared proof_system module (in namespace `bonk`) independent of `plonk` or `honk`. This makes sense as a goal for clarity of organization, but I am embarked on this to make it easier to incrementally rework the composer helper library in another PR. This PR accomplishes that goal. Along the way, I have done the following:
- Moved the `ComposerType` enum into `proof_system`.
- Anticipating my next PR, I split the composer helper library into two, with two firmly `plonk`-specific functions being moved into `plonk`. One or two will come out to `honk` later.
-  Anticipating my next PR, I moved what were `bonk::proving_key` and `bonk::verification_key` into `plonk`, since we will not share these structures with my forthcoming work.
- While debugging a linker error, I removed several instances of `using namespace bonk` in header files.
- To solve a linker error of the sort that I and @ledwards2225  had run into several times recently, I moved all shared composer helper library functions into the corresponding header file.
- To avoid repeatedly adding `bonk::` qualifiers, I have renamed the `bonk` namespace to the more boring and more reasonable `proof_system`, and I have nested `plonk` and `honk` inside of `proof_system`, so that when you're working side of `plonk` and `honk` there is no need to use the `bonk` qualifier. I'm open to suggestions for other solutions.  **This accounts for a huge proportion of the diff.**
- Resolved several comments about code being in bad namespaces.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/barretenberg/honk/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-barretenberg_module(honk numeric ecc srs proof_system transcript)
+barretenberg_module(honk numeric ecc srs proof_system transcript plonk)
```
Comment (ledwards2225) [inline cpp/src/barretenberg/honk/CMakeLists.txt:2]: I'm assuming this is temporary for the proving key or something. Worth leaving a TODO or is this part of your immediately forthcoming work?

File: cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.hpp [hunk 2]
```diff
@@ -2,45 +2,46 @@
...
     // TODO(#218)(kesha): we need to put this into the commitment key, so that the composer doesn't have to handle srs
     // at all
-    std::shared_ptr<bonk::ReferenceStringFactory> crs_factory_;
+    std::shared_ptr<ReferenceStringFactory> crs_factory_;
     bool computed_witness = false;
     StandardHonkComposerHelper()
-        : StandardHonkComposerHelper(
-              std::shared_ptr<bonk::ReferenceStringFactory>(new bonk::FileReferenceStringFactory("../srs_db/ignition")))
+        : StandardHonkComposerHelper(std::shared_ptr<ReferenceStringFactory>(
+              new proof_system::FileReferenceStringFactory("../srs_db/ignition")))
```
Comment (ledwards2225) [inline cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.hpp:35]: We should use `make_shared` here. Irrelevant to this PR

File: cpp/src/barretenberg/join_split_example/proofs/join_split/join_split_js_parity.test.cpp [hunk 3]
```diff
@@ -13,8 +13,8 @@ namespace proofs {
 namespace join_split {
 
 using namespace barretenberg;
-// using namespace plonk::stdlib::types;
-using namespace plonk::stdlib::merkle_tree;
+// using namespace proof_system::plonk::stdlib::types;
```
Comment (ledwards2225) [inline cpp/src/barretenberg/join_split_example/proofs/join_split/join_split_js_parity.test.cpp:16]: I do appreciate your unwavering commitment to the scope of this PR

File: cpp/src/barretenberg/plonk/proof_system/widgets/transition_widgets/arithmetic_widget.hpp [hunk 4]
```diff
@@ -32,13 +32,12 @@ template <class Field, class Getters, typename PolyContainer> class ArithmeticKe
...
   public:
-    inline static std::set<bonk::PolynomialIndex> const& get_required_polynomial_ids()
+    inline static std::set<PolynomialIndex> const& get_required_polynomial_ids()
     {
-        static const std::set<bonk::PolynomialIndex> required_polynomial_ids = {
-            bonk::PolynomialIndex::Q_1, bonk::PolynomialIndex::Q_2, bonk::PolynomialIndex::Q_3,
-            bonk::PolynomialIndex::Q_M, bonk::PolynomialIndex::Q_C, bonk::PolynomialIndex::W_1,
-            bonk::PolynomialIndex::W_2, bonk::PolynomialIndex::W_3
-        };
+        static const std::set<PolynomialIndex> required_polynomial_ids = { PolynomialIndex::Q_1, PolynomialIndex::Q_2,
```
Comment (ledwards2225) [inline cpp/src/barretenberg/plonk/proof_system/widgets/transition_widgets/arithmetic_widget.hpp:37]: pleasant.

File: cpp/src/barretenberg/proof_system/types/composer_type.hpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+namespace proof_system {
```
Comment (ledwards2225) [inline cpp/src/barretenberg/proof_system/types/composer_type.hpp:3]: Again, assuming this is to be addressed imminently and does not really require a TODO?

**Review by ledwards2225 — APPROVED** [review]
Looks good to me

File: cpp/src/barretenberg/honk/CMakeLists.txt (hunk 1 above)
Comment (codygunton) [reply cpp/src/barretenberg/honk/CMakeLists.txt:2]: Yeah, I'll make a note of that, thanks.

File: cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.hpp (hunk 2 above)
Comment (codygunton) [reply cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.hpp:35]: I'll make a note

**Pull request merged**